core/api/src/domain/phone-provider/index.types.d.ts

```diff
@@ -5,6 +5,7 @@ type PhoneCodeInvalidError = import("./errors").PhoneCodeInvalidError
 
 interface IPhoneProviderService {
   getCarrier(phone: PhoneNumber): Promise<PhoneMetadata | PhoneProviderServiceError>
+  validateDestination(phone: PhoneNumber): Promise<true | ApplicationError>
```

Comment:
this can not use ApplicationError, please use service error (in this case `PhoneProviderServiceError`)